autocomplete_light/fields.py

```diff
@@ -30,11 +27,11 @@ def __init__(self, autocomplete=None, registry=None, widget=None,
                     autocomplete_js_attributes, extra_context)
         kwargs['widget'] = widget
 
-        parents = super(FieldBase, self).__self_class__.__bases__
-        if ((forms.ModelChoiceField in parents or
-                forms.ModelMultipleChoiceField in parents)
-                and isinstance(self.autocomplete.choices, QuerySet)):
-            kwargs['queryset'] = self.autocomplete.choices
+        if 'queryset' not in kwargs:
+            parents = super(FieldBase, self).__self_class__.__bases__
+            if (forms.ModelChoiceField in parents or
+                    forms.ModelMultipleChoiceField in parents):
+                kwargs['queryset'] = self.autocomplete.choices
```

Comment:
Well I really dislike how I coupled Model stuff into FieldBase, can't find out what was the reason for that ... so should we consider moving that part into ModelChoiceField ?